src/Nethermind/Nethermind.JsonRpc/Modules/Admin/PeerInfo.cs

```diff
@@ -51,5 +54,11 @@ public PeerInfo(Peer peer, bool includeDetails)
                 LastSignal = (peer.InSession ?? peer.OutSession)?.LastPingUtc.ToString(CultureInfo.InvariantCulture);
             }
         }
+
+        private string CalculateClientId(PublicKey nodeKey)
+        {
+            byte[] publicKeyBytes = nodeKey.Bytes;
+            return (publicKeyBytes is null ? Keccak.Zero : Keccak.Compute(publicKeyBytes)).ToString(false);
```

Comment:
We can avoid allocation with:

```suggestion
            return (publicKeyBytes is null ? Keccak.Zero.ValueHash256 : ValueKeccak.Compute(publicKeyBytes)).ToString(false);
```